### PR TITLE
Replace doctrine/inflector with simpler mapping function

### LINKED ISSUE #8252: doctrine reflector dependency
# Bug Report

we depend on doctrine/reflector and make use of it only in a very few places.

I would question whether its worth the dependency or whether we can either use a simplified implementation or even drop the "inflector"-feature from the rules involved.

wdyt?

### PR DESCRIPTION
**Pull request by staabm**
closes https://github.com/rectorphp/rector/issues/8252

- `RectorNamingInflector` was only used in a single class, but we have `InflectorSingularResolver` with same functionality. Use `InflectorSingularResolver` everywhere
- use a simple english words singularizer - thats the only feautre we used from doctrine/inflector
- drop doctrine/inflector dependency

### CODE REVIEW

**Comment by staabm** [comment]
@TomasVotruba opinions? :-)

File: rules/Naming/ExpectedNameResolver/InflectorSingularResolver.php [hunk 1]
```diff
@@ -106,4 +100,83 @@ private function singularizeCamelParts(string $currentName): string
...
+            '/([lr])ves$/i' => '\\1f',
+            '/(tive)s$/i' => '\\1',
+            '/(hive)s$/i' => '\\1',
+            '/([^f])ves$/i' => '\\1fe',
+            '/(^analy)ses$/i' => '\\1sis',
+            '/((a)naly|(b)a|(d)iagno|(p)arenthe|(p)rogno|(s)ynop|(t)he)ses$/i' => '\\1\\2sis',
+            '/([ti])a$/i' => '\\1um',
+            '/(n)ews$/i' => '\\1ews',
+            '/s$/i' => '',
+        ];
```
Comment (TomasVotruba) [inline rules/Naming/ExpectedNameResolver/InflectorSingularResolver.php:133]: How does this compete with inflector setup? Some difference comparison would be helpful :+1:

File: rules/Naming/ExpectedNameResolver/InflectorSingularResolver.php [hunk 2]
```diff
@@ -37,11 +36,6 @@ final class InflectorSingularResolver
      */
     private const CAMELCASE = 'camelcase';
 
-    public function __construct(
-        private readonly Inflector $inflector
-    ) {
-    }
-
     public function resolve(string $currentName): string
```
Comment (TomasVotruba) [inline rules/Naming/ExpectedNameResolver/InflectorSingularResolver.php:39]: As new logic is added here, we'll need a test case to cover at lest few cases.

**Comment by staabm** [comment]
I have added the english test-suite from doctrine/inflector - except a few cases which conflict [with pre-existing tests](https://github.com/rectorphp/rector-src/blob/5ac3cb377abf968396c23573d8562e6ab6967f2a/tests/Naming/ExpectedNameResolver/InflectorSingularResolverTest.php#L30-L39).

File: rules/Naming/ExpectedNameResolver/InflectorSingularResolver.php (hunk 1 above)
Comment (staabm) [reply rules/Naming/ExpectedNameResolver/InflectorSingularResolver.php:133]: tbh, I am not convinved whether rector should do this pluralization/singularization stuff at all.

but at least its now reduced to a minimum

**Comment by TomasVotruba** [comment]
Thanks for the update and test coverage :+1: Let's ship this :wink:

**Comment by TomasVotruba** [comment]
I'm curious how much doctrine deps we'll drop :pray:

**Comment by TomasVotruba** [comment]
Looks good - https://github.com/rectorphp/rector/commit/52923dc606754a640fd75642fa8c28573715397d#diff-5ae063baecc4813da5c3f8e177d883ea93596a4c6abc8e5a2eaad7204923ed5f

**Comment by staabm** [comment]
nice <3.

**Comment by samsonasik** [comment]
@TomasVotruba the commit and tag seems overlapped:

- first commit have 0.18.6 tag : https://github.com/rectorphp/rector/commit/52923dc606754a640fd75642fa8c28573715397d
- next commit 0.18.6 tag : https://github.com/rectorphp/rector/commit/02041b220704b9cbe075f0310d0954b2fda5c40c (rolled back vendor/doctrine/inflector)

Run composer update on `^0.18.6` got vendor structure for doctrine inflector:

```
➜  ErrorHeroModule git:(master) ✗ tree vendor/rector/rector/vendor/doctrine/ -L 2         

vendor/rector/rector/vendor/doctrine/
└── inflector
    ├── LICENSE
    ├── README.md
    ├── composer.json
    └── lib
```

**Comment by samsonasik** [comment]
It seems because of the merge process is less than 20 seconds wait after tag 

https://github.com/rectorphp/rector-src/blob/1d3fc91bd5e0154daae34b1efdb64a6f78f47c89/.github/workflows/build_scoped_rector.yaml#L31

so it uses last commit that unrelated to tag, then overlapped after then, then, make 2 commits with tag 0.18.6

- first commit have 0.18.6 tag : https://github.com/rectorphp/rector/commit/52923dc606754a640fd75642fa8c28573715397d
- next commit 0.18.6 tag : https://github.com/rectorphp/rector/commit/02041b220704b9cbe075f0310d0954b2fda5c40c (rolled back vendor/doctrine/inflector)

**Comment by TomasVotruba** [comment]
@samsonasik  This is not tagged yet on purpose, it will be in 0.18.7

**Comment by samsonasik** [comment]
@staabm it unfortunatelly make invalid change on run rectify

change `class` to `clas` seems wrong:

```diff
-childClassReflection
+childClasReflection
```

see rectify PR on:

- https://github.com/rectorphp/rector-src/pull/5197#pullrequestreview-1696603640

**Comment by staabm** [comment]
I will have a look

**Pull request merged**